File: Code/Problem 7 - 10001st prime/nthPrime.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];

rl.on('line', (line) => {
    input.push(line);
});

rl.on('close', () => {
    main(input);
});

function main(input) {
    let t = parseInt(input.shift());
    for(let a0 = 0; a0 < t; a0++){
        let n = parseInt(input.shift());
        console.log(nthPrime(n)); //print test function
    }
}

//--------------------Code--------------------

function sieveOfTheotostenes(num) {
  let sieve = Array(num+1).fill(true);
  let primes = [];

  for (let i = 2; i*i < num; i++) {
    if (sieve[i]) {
      for (let j = i; j*i < num; j++) {
        sieve[j*i] = false;
      }
    }
  }

  for (let i = 2; i < sieve.length; i++) {
    if (sieve[i]) primes.push(i);
  }

  return primes;
}

const primes = sieveOfTheotostenes(104743);

function nthPrime(n) {
  return primes[n-1];
}

File: Code/Problem 7 - 10001st prime/NthPrime.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class NthPrime {
    public static List<Integer> sieveOfErotosthenes(int num) {
        boolean[] sieve = new boolean[num+1];
        List<Integer> primes = new ArrayList<>();
        
        Arrays.fill(sieve, true);
        
        for (int i = 2; i*i < sieve.length; i++) {
            if (sieve[i]) {
                for (int j = i; j*i < sieve.length; j++) {
                    sieve[j*i] = false;
                }
            }
        }
        
        for (int p = 2; p < sieve.length; p++) {
            if (sieve[p]) primes.add(p);
        }
        
        return primes;
    }

    public static void main(String[] args) {
        
        List<Integer> primes = sieveOfErotosthenes(104743);
        
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            int n = in.nextInt();
            System.out.println(primes.get(n-1));
        }
    }
}